qemu/tests/nic_hotplug.py

```diff
@@ -122,16 +133,32 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
             nic_model = nic_params["pci_model"]
             nic_params["nic_model"] = nic_model
             nic_params["nic_name"] = nic_name
+            used_sameid = params.get("used_sameid", "no")
 
-            error.context("Disable other link(s) of guest", logging.info)
-            disable_nic_list = primary_nic + nic_hotplugged
-            for nic in disable_nic_list:
-                if guest_is_linux:
-                    ifname = utils_net.get_linux_ifname(s_session, nic["mac"])
-                    s_session.cmd_output_safe("ifconfig %s 0.0.0.0" % ifname)
-                else:
-                    s_session.cmd_output_safe("ipconfig /release all")
-                vm.set_link(nic.device_id, up=False)
+            if used_sameid == "yes":
+                useddevice_id = primary_nic[0].netdev_id
+                error.context("Hot-plug NIC with the netdev already in use",
+                              logging.info)
+                try:
+                    device_add_nic(nic_model, useddevice_id, nic_name)
+                except Exception, err_msg:
+                    match_error = "it's in use"
+                    if match_error in str(err_msg):
+                        pass
+                    else:
```

Comment:
I think this 'else' should be moved back, align to 'except', which means you do not catch any exceptions, then the nic is plugged.
Like:
try:
    device_add_nic(nic_model, useddevice_id, nic_name)
except Exception, err_msg:
    match_error = "it's in use"
    if match_error in str(err_msg):
        pass
    else:
        raise error.TestFail("Error message does not match")
else:
     raise error.TestFail("Test failed, %s can hot plug NIC when\
                                it's already in use" % useddevice_id)